docs/source/architecture/transactions_and_batches.rst

```diff
@@ -0,0 +1,189 @@
+
+
+.. note::
+
+   The original header bytes as constructed from the sender are used
+   for verification of the signature.  It is not considered good practice to
+   de-serialize the header (for example, to a Python object) and then
+   re-serialize the header with the intent to produce the same byte sequence as
+   the original.  Serialization can be sensitive to programming language or
+   library, and any deviation would produce a sequence that would not match the
+   signature; thus, best practice is to alway use the original header bytes for
+   verification.
+
+Transaction Family
+------------------
+
+In Sawtooth Lake, the set of possible transactions are defined by an extensible
+system called transaction families.  Defining and implementing a new
+transaction family adds to the taxonomy of available transactions which can be
+applied. For example, in the :ref:`transaction-family-tutorial`, we
+define a transaction family called "xo" which defines a set of transactions for
+playing tic-tac-toe.
+
+In addition to the name of the transaction family (family_name), each
+transaction specifies a family version string (family_version).  The
+payload_encoding field allows the transaction family to support multiple
+serialization mechanisms (for example, json or cbor).
+
+Dependencies and Input/Output Addresses
+---------------------------------------
+
+Transactions can depend upon other transactions, which is to say a dependent
+transaction cannot be applied prior the transaction upon which it depends.
+
+The dependencies field of a transaction allows explicitly specifying the
+transactions which must be applied prior to the current transaction.  Explicit
+dependencies are useful in situations where transactions have dependencies but
+can not be placed in the same batch (for example, if the transactions are
+submitted at different times).
+
+To assist in parallel scheduling operations, the inputs and outputs fields of a
+transaction contain state addresses.  The scheduler determines the implicit
+dependencies between transactions based on interaction with state.  The
+addresses may be fully qualified leaf-node addresses or partial prefix
+addresses.  Input addresses are read from the state and output addresses are
+written to state.  While they are specified by the client, input and output
+declarations on the transaction are enforced during transaction execution.
+Partial addresses work as wildcards and allow transactions to specify parts of
+the tree instead of just leaf nodes.
+
+Payload
+-------
+
+The payload is used during transaction execution as a way to convey the change
+which should be applied to state.  Only the transaction family processing the
+transaction will deserialize the payload; to all other components of the
+system, payload is just a sequence of bytes.  The transaction family processing
+the transaction will deserialize the bytes using payload_encoding as a hint to
+determine the appropriate technique.
```

Comment:
Maybe clarify here that the transaction family itself is responsible for defining the payload_encoding.